master/buildbot/util/httpclientservice.py

```diff
@@ -175,25 +176,25 @@ def readContent(session, res):
             if self.debug:
                 log.debug("==> {code}: {content}", code=res.status_code, content=res.content)
             return res
+
         # read the whole content in the thread
         kwargs['background_callback'] = readContent
         if self.verify is False:
             kwargs['verify'] = False
-        d = self._session.request(method, url, **kwargs)
-        d.addCallback(TxRequestsResponseWrapper)
-        d.addCallback(IHttpResponse)
-        return d
 
+        res = yield self._session.request(method, url, **kwargs)
+        return IHttpResponse(TxRequestsResponseWrapper(res))
```

Comment:
this breaks on python2. You need to use defer.returnValue inside inlineCallbacks

Reply:
Will that matter though?

Reply:
right. We can start removing defer.returnValue() from the codebase after we drop 2.7

Reply:
Will change